llvm/test/CodeGen/SPIRV/hlsl-intrinsics/SV_DispatchThreadID.ll

```diff
@@ -0,0 +1,76 @@
+; RUN: llc -O0 -mtriple=spirv-vulkan-unknown %s -o - | FileCheck %s
+; RUN: %if spirv-tools %{ llc -O0 -mtriple=spirv-vulkan-unknown %s -o - -filetype=obj | spirv-val %}
+
+; This file generated from the following HLSL:
+; clang -cc1 -triple spirv-vulkan-library -x hlsl -emit-llvm -disable-llvm-passes -finclude-default-header -o - DispatchThreadID.hlsl
+;
+; [shader("compute")]
```

Comment:
Meeting follow-up: It was suggested that instead of this LLVM IR test, we could instead have a RUN line that references a `.hlsl` test file from frontend tests so that it would also catch changes in the HLSL->LLVM IR path, since this backend tests is so dependent on that translation.

Reply:
I don't know the context of that discussion, but I think the change made here is a backend change transforming IR to SPIR-V. It seems to me that the test should be IR->SPIR-V as well.

LLVM's test suite does not depend on building clang, so you can't use a clang run line in an LLVM test, and this code (as an LLVM change) should have a test in LLVM.

Reply:
We'll have a lot of cases like this when more and more intrinsics are added.
Is there a way to share intrinsics between spirv and dxil instead of duplicate 
 those intrinsics?

Reply:
Is it possible for DXIL and SPIRV to share the same backend target as different sub targets?

Reply:
The LLVM Offload project has been using library calls instead of intrinsics where it makes sense to reduce the need to duplicate intrinsics. We can do that as well, but in cases like this specific change I think having target specific intrinsics has minimal cost and simplifies the lowering path.

There is no reasonable way to have DXIL and SPIR-V share a target. They are wildly different and I suspect any abstraction layers we would need to add to make that reasonable would be more complex than they are worth.